src/solver/td3.ml

```diff
@@ -319,17 +340,24 @@ module Base =
           in
           HM.remove called x;
           let old = HM.find rho x in (* d from older solve *) (* find old value after eq since wpoint restarting in eq/eval might have changed it meanwhile *)
+
+          (* if value has changed, reduce gas (only applies to marked widening points) *)
+          if not (term && phase = Narrow) && not (S.Dom.equal eqd old) then reduce_gas x;
+
           let wpd = (* d after widen/narrow (if wp) *)
             if not wp then eqd
-            else if term then
-              match phase with
-              | Widen -> S.Dom.widen old (S.Dom.join old eqd)
-              | Narrow when GobConfig.get_bool "exp.no-narrow" -> old (* no narrow *)
-              | Narrow ->
-                (* assert S.Dom.(leq eqd old || not (leq old eqd)); (* https://github.com/goblint/analyzer/pull/490#discussion_r875554284 *) *)
-                S.Dom.narrow old eqd
-            else
-              box old eqd
+            else (
+              if term then
+                match phase with
+                | Widen -> S.Dom.widen old (S.Dom.join old eqd)
+                | Narrow when GobConfig.get_bool "exp.no-narrow" -> old (* no narrow *)
+                | Narrow ->
+                  (* assert S.Dom.(leq eqd old || not (leq old eqd)); (* https://github.com/goblint/analyzer/pull/490#discussion_r875554284 *) *)
+                  S.Dom.narrow old eqd
+              else (
+                box old eqd
+              )
+            )
```

Comment:
This seems to be just changing the formatting which would be nice to not change unnecessarily.